metpy/calc/thermo.py

```diff
@@ -2054,3 +2054,89 @@ def static_stability(pressure, temperature, axis=0):
 
     return - Rd * temperature / pressure * first_derivative(np.log(theta / units.K),
                                                             x=pressure, axis=axis)
+
+
+@exporter.export
+@check_units('[pressure]', '[temperature]', '[temperature]', '[pressure]')
+def downdraft_cape(pressure, temperature, dewpoint, parcel=None, bottom=None,
+                   top_pressure=400 * units.hPa):
+    r"""Calculate downdraft CAPE.
+
+    Calculate the downdraft convective available potential energy (CAPE).The minimum theta-e
+    value between the surface and top_pressure is used as the parcel starting point. Parcels
+    descend along a moist adiabat. Area between the parcel path and environmental temperature
+    is integrated to calculate DCAPE.
+
+    Parameters
+    ----------
+    pressure : `pint.Quantity`
+        The atmospheric pressure level(s) of interest. The first entry should be the starting
+        point pressure.
+    temperature : `pint.Quantity`
+        The atmospheric temperature corresponding to pressure.
+    dewpoint : `pint.Quantity`
+        The atmospheric dew point corresponding to pressure.
+    parcel : tuple
+        Tuple of pressure and temperature for the starting parcel.
+    bottom : `pint.Quantity`
+        Lowest point in the parcel path to consider in integration in height or pressure.
+        If no heights are given, a standard atmosphere will be assumed. Defaults to None.
+    top_pressure : `pint.Quantity`
+        The lowest pressure value to be considered in the calculation. Defaults to 400 hPa.
+
+    Returns
+    -------
+    `pint.Quantity`
+        Downdraft convective available potential energy (CAPE).
+
+    Notes
+    -----
+    .. math:: \text{DCAPE} = R_d \int_{P_i}^{P_{sfc}} (T_{env} - T_{parcel}) d\text{ln}(p)
+
+    * :math:`DCAPE` Downdraft convective available potential energy
+    * :math:`R_d` Gas constant
+    * :math:`g` Gravitational acceleration
+    * :math:`T_{parcel}` Parcel temperature
+    * :math:`T_{env}` Environment temperature
+    * :math:`p` Atmospheric pressure
+
+    """
+    # Sort so everything is with monotonically increasing pressure.
+    sort_inds = np.argsort(pressure)
+    pressure = pressure[sort_inds]
+    temperature = temperature[sort_inds]
+    dewpoint = dewpoint[sort_inds]
+
+    # Trim data to only top height and below as well as bottom height and above
+    pressure, temperature, dewpoint = get_layer(pressure, temperature, dewpoint,
```

Comment:
This call of `get_layer` only differs from the one below with the passing of `bottom`. Are they both needed?

Reply:
That was a try at something that won't survive 😄 I think I'm going to refactor some here - we don't want to trim before we interpolate a parcel.... hmmm.. more to come.